### PR TITLE
fix: Fixes #18519 - makes DefinePlugin quieter under default log level

### LINKED ISSUE #18519: Too many logs by DefinePlugin
# Bug report

**What is the current behavior?**

This is how most of my build logs look after #15778 was fixed

```
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
    Replaced "process.env.NODE_ENV" with ""development""
```

If the text above frustrated you, I really appreciate you understanding my frustration. 

**If the current behavior is a bug, please provide the steps to reproduce.**

**What is the expected behavior?**

Less verbose output, less repetition, concise and impactful logs. This feels like a great candidate for using `logger.debug` instead.

**Other relevant information:**
webpack version: `5.89.0`
Node.js version: `18.13.0`
Operating System: `Mac Sonoma on M1 pro`
Additional tools: `¯\_(ツ)_/¯`

### PR DESCRIPTION
**Pull request by sushruth**
Fixes #18519 

**What kind of change does this PR introduce?**

Technically a bugfix but more like removing an annoying log.

**Did you add tests for your changes?**

~~This does not change any functionality and there is no test to check for verbosity of logs.~~ I was wrong. I am fixing related tests right now.

**Does this PR introduce a breaking change?**

No.

**What needs to be documented once your changes are merged?**

Default logging behavior of DefinePlugin - but that is already not documented, so no new behavior from the perspective of documentation. It would definitely be nice to add that to the docs.

### CODE REVIEW

**Review by alexander-akait — CHANGES_REQUESTED** [review]
> logger.info(...): for important information messages. These messages are displayed by default. Only use this for messages that the user really needs to see

> logger.log(...): for unimportant information messages. These messages are displayed only when user had opted-in to see them

So it works as expected

**Comment by sushruth** [comment]
@alexander-akait I would argue that this definition fits well.

> logger.debug(...): for debugging information. These messages are displayed only when user had opted-in to see debug logging for specific modules

Repeating that one line of log with no other information or context to go with it, is causing more noise than giving useful information. I think if that line of log said where the code was replaced while debugging DefinePlugin, that would at least amount to useful information. 

I would also argue that if we definitely need a log, webpack must display one unique log line item per replaced variable for `logger.log` level while `logger.debug` could be used to say which exact file and what line it happens at. That way debugging is easier and regular operation is not aggressively verbose.

What are your thoughts on this?

**Comment by alexander-akait** [comment]
I am fine with `logger.debug`

**Comment by alexander-akait** [comment]
And other improvements

File: lib/DefinePlugin.js [hunk 1]
```diff
@@ -272,7 +272,7 @@ const toCode = (
 
 	const strCode = transformToCode();
 
-	logger.log(`Replaced "${key}" with "${strCode}"`);
+	logger.info(`Replaced "${key}" with "${strCode}"`);
```
Comment (alexander-akait) [inline lib/DefinePlugin.js:275]: Please use `debug`

Comment (sushruth) [reply lib/DefinePlugin.js]: Done.

**Comment by sushruth** [comment]
> And other improvements

I will try to open a separate PR for it since it involves a bit more logic. Thanks, @alexander-akait !

**Review by alexander-akait — APPROVED** [review]

**Pull request merged**